### PR TITLE
Add support for HTTPS

### PR DESCRIPTION
**Pull request by maks**
Adds HTTPS support to dhttpd.

Thanks to the original author of PR #32  that this is based on.

Fixes: #1

### CODE REVIEW

**Comment by maks** [comment]
@kevmoo when you get a chance, would you be able to approve running the workflows for this PR please.

**Comment by kevmoo** [comment]
@maks - still some tests to fix!

File: lib/dhttpd.dart [hunk 1]
```diff
@@ -19,6 +20,8 @@ class Dhttpd {
 
   String get urlBase => 'http://$host:$port/';
 
+  static bool get isSSL => _ssl;
```
Comment (kevmoo) [inline lib/dhttpd.dart:23]: Make this a normal final field and pass it in.

Comment (maks) [reply lib/dhttpd.dart:23]: Doh, yes sorry @kevmoo ! I did this is a bit of a rush and didnt properly review the code from the previous PR that I cherry picked.

File: lib/src/options.dart [hunk 2]
```diff
@@ -34,6 +34,30 @@ class Options {
...
+      valueHelp: 'sslkey',
+      help: 'The key of the SSL certificate to use.'
+          '\r\nIf set along with sslcert, https will be used.'
+          '\r\nSee the dart documentation about '
+          'SecurityContext.usePrivateKey for more.')
+  final String? sslkey;
+
+  @CliOption(
+      valueHelp: 'sslkeypassword',
+      help: 'The password for the key of the SSL certificate to use.'
```
[the remark this answers is not in the document]
Comment (maks) [reply lib/src/options.dart:55]: 100% again sorry for not spotting that when I redid the old PR.

[the remark this answers is not in the document]
Comment (kevmoo) [reply lib/src/options.dart:55]: Zero worries, yo. That's why we have PRs!

**Pull request merged**